src/include/duckdb/storage/table/row_group.hpp

```diff
@@ -184,29 +189,31 @@ class RowGroup : public SegmentBase<RowGroup> {
 	idx_t GetRowGroupSize() const;
 
 private:
-	optional_ptr<RowVersionManager> GetVersionInfo();
+	optional_ptr<RowVersionManager> GetVersionInfo() const;
 	shared_ptr<RowVersionManager> GetOrCreateVersionInfoPtr();
 	shared_ptr<RowVersionManager> GetOrCreateVersionInfoInternal();
-	void SetVersionInfo(shared_ptr<RowVersionManager> version);
+	void SetVersionInfo(shared_ptr<RowVersionManager> version) const;
 
+	const ColumnData &GetColumn(storage_t c) const;
+	const ColumnData &GetColumn(const StorageIndex &c) const;
 	ColumnData &GetColumn(storage_t c);
 	ColumnData &GetColumn(const StorageIndex &c);
 	idx_t GetColumnCount() const;
 	vector<shared_ptr<ColumnData>> &GetColumns();
 
 	template <TableScanType TYPE>
-	void TemplatedScan(TransactionData transaction, CollectionScanState &state, DataChunk &result);
+	void TemplatedScan(TransactionData transaction, CollectionScanState &state, DataChunk &result) const;
 
 	vector<MetaBlockPointer> CheckpointDeletes(MetadataManager &manager);
 
 	bool HasUnloadedDeletes() const;
 
 private:
-	mutex row_group_lock;
+	mutable mutex row_group_lock;
 	vector<MetaBlockPointer> column_pointers;
 	unique_ptr<atomic<bool>[]> is_loaded;
 	vector<MetaBlockPointer> deletes_pointers;
-	atomic<bool> deletes_is_loaded;
+	mutable atomic<bool> deletes_is_loaded;
```

Comment:
Same as above, maybe we can get around `mutable` by using `load()`

Reply:
```c++
optional_ptr<RowVersionManager> RowGroup::GetVersionInfo() const {
	if (!HasUnloadedDeletes()) {
		// deletes are loaded - return the version info
		return version_info;
	}
	lock_guard<mutex> lock(row_group_lock);
	// double-check after obtaining the lock whether or not deletes are still not loaded to avoid double load
	if (!HasUnloadedDeletes()) {
		return version_info;
	}
	// deletes are not loaded - reload
	auto root_delete = deletes_pointers[0];
	auto loaded_info = RowVersionManager::Deserialize(root_delete, GetBlockManager().GetMetadataManager(), start);
	SetVersionInfo(std::move(loaded_info));
	deletes_is_loaded = true;
	return version_info;
}
```

<https://github.com/duckdb/duckdb/pull/8869>